lab-taylor/lib/client-pool.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const commands = require('./commands');
+
+function generateNickname() {
+  const nicknameBase = ['monkey', 'slug', 'hippo', 'yeti', 'unicorn'];
+  const idx = Math.floor(Math.random() * ( nicknameBase.length - 0));
+  const num = Math.floor(Math.random() * (99 - 0));
+  return `${nicknameBase[idx]}-${num}`;
+}
+
+function registerClient(socket, clientPool){
+  socket.wack = {};
+  socket.wack.id = 'user_' + Date.now();
+  socket.wack.nickname = generateNickname();
+  clientPool.pool[socket.wack.id] = socket;
+}
+
+function broadcast(data, socket, clientPool) {
+  Object.keys(clientPool.pool).forEach((clientId) => {
+    clientPool.pool[clientId].write(`${socket.wack.nickname}:: ${data.toString()}`);
+  });
+}
+
+function parseCommands(data, socket, clientPool) {
+  const commandMap = {
+    '\\nick': commands.updateNickname,
+    '\\dogbomb\r\n': commands.doggie
+  };
+
+  broadcast(data, socket, clientPool);
+
+  if (data.toString()[0] === '\\') {
```

Comment:
I like how you've done this, in my experience though if a wack command is used it generally won't show up in chat, rather the command output/action will just be completed.  Here it would be like putting the previous broadcast function call into an else block after the if statement on line 34.  Nothing wrong with your implementation though, just a wack comment.